shell/plugins/plugin.js

```diff
@@ -26,38 +27,69 @@ export default async function(context) {
   }
 
   if (loadPlugins) {
-    // TODO: Get rancher version using the new API (can't use setting as we have not loading the store)
-    const rancherVersion = undefined;
+    let rancherVersion;
+    let kubeVersion;
+
+    const reqs = [];
+
+    // Fetch rancher version metadata
+    reqs.push(context.store.dispatch('rancher/request', {
+      url:                  '/rancherversion',
+      method:               'get',
+      redirectUnauthorized: false
+    }));
+
+    // Fetch kubernetes version metadata
+    reqs.push(context.store.dispatch('rancher/request', {
+      url:                  '/version',
```

Comment:
We need to bear in mind both this and /rancherversion could always fail on some systems (if not already) given the advise given out in [docs](https://documentation.suse.com/external-tree/en-us/cloudnative/rancher-manager/v2.9/en/security/rancher-security-best-practices.html#_restrict_public_access_to_version_and_rancherversion_path) (we were not consulted). To confirm if either are missing this process won't fall over, we just might load extensions we shouldn't?

Reply:
Unfortunately, through other places, I was already aware of that 😢 

Since the script is based on a `Promise.allSettled`, as long as 

```
reqs.push(context.store.dispatch('management/request', {
      url:                  `${ UI_PLUGIN_BASE_URL }`,
      method:               'GET',
      headers:              { accept: 'application/json' },
      redirectUnauthorized: false,
    }));
```
doesn't fail, extensions are still loaded. The side-effect is that we won't be able to prevent loading of extensions that are already installed based on `rancherVersion` and `kubeVersion`.

Bear in mind that if `/rancherversion` get blocked, we have the `prime` mechanism that get's broken as well and that's much more important, imo 😛